### ISSUE TITLE
4.15.0 Unable to start or stop containers from GUI (CLI is okay)

### ISSUE THREAD

**Issue by x0rsw1tch** [docker/for-win#13057]
- [ x ] I have tried with the latest version of Docker Desktop
  - [ x ] I have tried disabling enabled experimental features
  - [ x ] I have uploaded Diagnostics
  - Diagnostics ID: 2EFD3B9E-5D4B-4939-9992-2DF137757D3F/20221108233204
  - 4.14.1 Diagnostic ID: 2EFD3B9E-5D4B-4939-9992-2DF137757D3F/20221124002607

### Actual behavior

When attempting to start containers, this message displays:

![image](https://user-images.githubusercontent.com/9543257/200700313-41b0e745-cd98-41a0-ab87-46ae2cda7e50.png)

When attempting to stop containers, this message displays:

![image](https://user-images.githubusercontent.com/9543257/200700188-6a1ff77d-23cc-4316-93c1-15a9d770cb5c.png)

### Expected behavior

Can start or stop containers via desktop GUI application.

### Information

  - Is it reproducible? Yes. It is a consistent problem since upgrading to 4.13.1
  - Is the problem new? Yes, this problem did not occur in previous versions
  - Did the problem appear with an update? Yes, when upgrading to 4.1.3.1 and to 4.14.1

  - Windows Version: Windows 10 Professional x64 21H2
  - Docker Desktop Version: 4.13.1
  - WSL2 or Hyper-V backend?  Hyper-V
  - Are you running inside a virtualized Windows e.g. on a cloud server or a VM: No

### (4.13.1) Output of `& "C:\Program Files\Docker\Docker\resources\com.docker.diagnose.exe" check`

```
[2022-11-08T23:48:43.236734200Z][com.docker.diagnose.exe][I] set path configuration to OnHost
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0002: does the bootloader have virtualization enabled?
[PASS] DD0020: is the Hyper-V Windows Feature enabled?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0013: is the $PATH ok?
[PASS] DD0003: is the Docker CLI working?
[PASS] DD0005: is the user in the docker-users group?
[FAIL] DD0014: are the backend processes running? 1 error occurred:
        * vpnkit.exe is not running


[PASS] DD0007: is the backend responding?
[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[PASS] DD0006: is the Docker Desktop Service responding?
[PASS] DD0012: is the VM networking working?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0033: does the host have Internet access?
[PASS] DD0002: does the bootloader have virtualization enabled?
[PASS] DD0020: is the Hyper-V Windows Feature enabled?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0032: do Docker networks overlap with host IPs?

Please investigate the following 1 issue:

1 : The test: are the backend processes running?
    Failed with: 1 error occurred:
        * vpnkit.exe is not running
```

### (4.14.1) Output of `& "C:\Program Files\Docker\Docker\resources\com.docker.diagnose.exe" check`

```
[2022-11-24T00:29:22.309858100Z][com.docker.diagnose.exe][I] set path configuration to OnHost
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0002: does the bootloader have virtualization enabled?
[PASS] DD0020: is the Hyper-V Windows Feature enabled?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0013: is the $PATH ok?
[PASS] DD0003: is the Docker CLI working?
[PASS] DD0005: is the user in the docker-users group?
[PASS] DD0014: are the backend processes running?
[PASS] DD0007: is the backend responding?
[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[PASS] DD0006: is the Docker Desktop Service responding?
[PASS] DD0012: is the VM networking working?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0033: does the host have Internet access?
[PASS] DD0002: does the bootloader have virtualization enabled?
[PASS] DD0020: is the Hyper-V Windows Feature enabled?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0032: do Docker networks overlap with host IPs?
No fatal errors detected.
```

### Steps to reproduce the behavior

1. Open desktop app
1. Attempt to start some containers that are orchestrated with docker-compose
1. Get error

### Other notes

No problems using docker-compose in CLI, or with the VSC docker extension

**Comment by ehill-iolani** [comment]
I ran into a similar yesterday when updating from Docker Desktop v4.12 to v4.14. In version v4.14 there is a "copy run command" option that shows the docker run command the GUI uses. Turns out the run command from the GUI is missing a `-t` flag. Inserting the `-t` appears to fix the issues. FYI I am running WSL2 on Windows 10 Home x64 22H2 but the issues appears to be within the Docker Desktop GUI itself:

Base run command from GUI:
```
docker run --hostname=d8b100024165 --env=PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin --runtime=runc -d ubuntu:latest
```

Modified run command that appears to fix the issue:
```
docker run --hostname=d8b100024165 --env=PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin --runtime=runc -t -d ubuntu:latest
```

I have not tried to edit the run command for the GUI but this seems to be the issue.

Let me know if this modification also works on your end. I'd be interested to hear if this works for you.

**Comment by ehill-iolani** [comment]
@hydreg In v4.14 there should be "copy docker run" option when you select the three circles from the right sidebar. I have attached an image below. You can then modify the docker run command in a text editor and throw it back into the CLI and then your container should work. FYI it does make a new container but this new container will be accessible from the GUI as normal; the start/stop/view details will all work.

![docker_run_command](https://user-images.githubusercontent.com/112661276/202571597-a6efe30b-2d27-4873-93f2-1cc35a5f9ee7.PNG)

**Comment by hydreg** [comment]
@ehill-iolani 
Oh, I see now, I had version 4.14.0 installed, while this functionality was introduced in 4.14.1, plus I've been looking at container kebab menu instead of service kebab menu. That doesn't help me much though, since running separate services within container works fine grom the GUI, it's containers that are broken. Thank you for your help though.

![Docker_Desktop_2022-11-22_08-27-00](https://user-images.githubusercontent.com/22574610/203251566-673b57ea-f3ab-4533-bd6e-f899f8e53a41.jpg)

**Comment by x0rsw1tch** [comment]
Problem persists after upgrading to 4.14.1. Diagnostic output is different this time, no errors were detected this time around. Uploaded a new diagnostic report.

**Comment by x0rsw1tch** [comment]
Problem still persists after upgrading to 4.15.0. Decided to backup my working containers and do a fresh install, and even after that, the problem is still there.

**Comment by x0rsw1tch** [comment]
Found that my custom `default-address-pools` config in `daemon.json` was gone, presumably delete overwritten by some previous upgrade, or during reinstall. Diagnostic tool started reported overlapping network pools between WSL and Docker. After adding a custom `default-address-pools` entry, the problem has disappeared for me.

For reference, this is the config I added

![image](https://user-images.githubusercontent.com/9543257/208547996-f6b25d02-8539-4e9e-98e0-a6d6ee28927b.png)

I have no idea why docker configured its default network pool CIDR range to be in direct conflict with another already established network. 

For posterity, this is what my networks look like on my PC before I fixed it:

```
Hyper-V    172.18.48.0/20
WSL        172.22.32.0/20
Docker     172.22.0.0/16
```

And after:

```
Hyper-V    172.18.48.0/20
WSL        172.22.32.0/20
Docker     172.27.0.0/16
```

In any case, I am now able to stop/start containers from the GUI now!

@hydreg I'm going to close this as your problem appears to be a bit different from the problem I was having. Assuming you're still having trouble, might want to spin up a new ticket.
